examples/delay_based_on_error_test.go

```diff
@@ -0,0 +1,108 @@
+// This test delay is based on kind of error
+// e.g. HTTP response [Retry-After](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After)
+package retry_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/avast/retry-go"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseRetryAfter(t *testing.T) {
+	retryAfter, err := ParseRetryAfter("xyz")
+	assert.Error(t, err)
+
+	retryAfter, err = ParseRetryAfter("")
+	assert.Error(t, err)
+
+	testDur := 120 * time.Second
+
+	retryAfter, err = ParseRetryAfter("120")
+	assert.NoError(t, err)
+	assert.Equal(t, testDur, retryAfter, "time in seconds")
+
+	retryAfter, err = ParseRetryAfter(time.Now().Add(testDur).Format(time.RFC850))
+	assert.NoError(t, err)
+	assert.True(t, retryAfter > (115*time.Second), "time in seconds are ~120")
+	t.Log(retryAfter)
+}
+
+func ParseRetryAfter(ra string) (time.Duration, error) {
+	if ra == "" {
+		return 0, fmt.Errorf("Retry-After header was empty")
+	}
+
+	t, errParse := http.ParseTime(ra)
+	if errParse != nil {
+		if sec, errParse := strconv.Atoi(ra); errParse == nil {
+			return time.Duration(sec) * time.Second, nil
+		}
+	} else {
+		return t.Sub(time.Now()), nil
+	}
+
+	return 0, fmt.Errorf("Invalid Retr-After format %s", ra)
+}
+
+type RetryAfterError struct {
+	response http.Response
+}
+
+func (err RetryAfterError) Error() string {
+	return fmt.Sprintf("Request to %s fail %s (%d)", err.response.Request.RequestURI, err.response.Status, err.response.StatusCode)
+}
+
+type SomeOtherError struct {
+	err        string
+	retryAfter time.Duration
+}
+
+func (err SomeOtherError) Error() string {
+	return err.err
+}
+
+func TestCustomRetryFunctionBasedOnKindOfError(t *testing.T) {
+	url := "http://example.com"
```

Comment:
I don't fully understand this file, I guess that 
- it feels wrong to use real http in test, httptest library is usually used
   - and `http` is in fact required to test the `retry function based on custom error`
   - parsing of HTTP Retry-After is out of scope of this library and shall not be tested
- SomeOtherError and RetryAfterError does seem to be used, they seem to be useless

Reply:
the idea of examples folder was real examples of retry-go usage... parsing of HTTP Retry-After is out of the scope of this library - I agree. But I needed a place to try it, and this wasn't a bad place - someone can be inspired by this.